thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/events/HolidayEventsLoader.java

```diff
@@ -0,0 +1,364 @@
+      this.name = name;
+      this.eventType = eventType;
+      this.startTime = startTime;
+      this.endTime = endTime;
+    }
+
+    /**
+     * Gets name.
+     *
+     * @return the name
+     */
+    public String getName() {
+      return name;
+    }
+
+    /**
+     * Sets name.
+     *
+     * @param name the name
+     */
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    /**
+     * Gets start time.
+     *
+     * @return the start time
+     */
+    public long getStartTime() {
+      return startTime;
+    }
+
+    /**
+     * Sets start time.
+     *
+     * @param startTime the start time
+     */
+    public void setStartTime(long startTime) {
+      this.startTime = startTime;
+    }
+
+    /**
+     * Gets end time.
+     *
+     * @return the end time
+     */
+    public long getEndTime() {
+      return endTime;
+    }
+
+    /**
+     * Sets end time.
+     *
+     * @param endTime the end time
+     */
+    public void setEndTime(long endTime) {
+      this.endTime = endTime;
+    }
+
+    /**
+     * Gets event type.
+     *
+     * @return the event type
+     */
+    public String getEventType() {
+      return eventType;
+    }
+
+    /**
+     * Sets event type.
+     *
+     * @param eventType the event type
+     */
+    public void setEventType(String eventType) {
+      this.eventType = eventType;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(getName(), getEventType(), getStartTime(), getEndTime());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof HolidayEvent)) {
+        return false;
+      }
+      HolidayEvent holidayEvent = (HolidayEvent) obj;
+      return Objects.equals(getName(), holidayEvent.getName()) && Objects.equals(getStartTime(),
+          holidayEvent.getStartTime()) && Objects.equals(getEndTime(), holidayEvent.getEndTime()) && Objects.equals(
+          getEventType(), holidayEvent.getEventType());
+    }
+  }
+
+  /**
+   * Instantiates a new Holiday events loader.
+   *
+   * @param holidayEventsLoaderConfiguration the configuration
+   * @param calendarApiKeyPath the calendar api key path
+   * @param eventDAO the event dao
+   */
+  public HolidayEventsLoader(HolidayEventsLoaderConfiguration holidayEventsLoaderConfiguration,
+      String calendarApiKeyPath, EventManager eventDAO) {
+    this.holidayRange = holidayEventsLoaderConfiguration.getHolidayRange();
+    this.calendarList = holidayEventsLoaderConfiguration.getCalendars();
+    this.keyPath = calendarApiKeyPath;
+    this.eventDAO = eventDAO;
+    scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+  }
+
+  /** List of google holiday calendar ids */
+  private List<String> calendarList;
+
+  /** Calendar Api private key path */
+  private String keyPath;
+  private ScheduledExecutorService scheduledExecutorService;
+  private TimeGranularity runFrequency = new TimeGranularity(7, TimeUnit.DAYS);
```

Comment:
This should be part of the configuration

Reply:
Sure. Will change.